src/view/MixerElementSelectorWindow.vala

```diff
@@ -0,0 +1,117 @@
+/*
+ * brickman -- Brick Manager for LEGO MINDSTORMS EV3/ev3dev
+ *
+ * Copyright (C) 2016 Kaelin Laundry <wasabifan@outlook.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * MixerElementSelectorWindow.vala - Lists ALSA mixer elements
+ */
+
+using Ev3devKit;
+using Ev3devKit.Ui;
+
+namespace BrickManager {
+
+    public class MixerElementSelectorWindow : BrickManagerWindow {
+        Ui.Menu element_menu;
+
+        public signal void mixer_element_selected (IMixerElementViewModel selected_element);
+
+        public MixerElementSelectorWindow () {
+            title = "Sound mixer elements";
+            element_menu = new Ui.Menu ();
+            content_vbox.add (element_menu);
+        }
+
+        protected string get_element_label_text (IMixerElementViewModel element) {
+            string mute_string = (element.can_mute && element.is_muted) ? ", muted" : "";
```

Comment:
I think it would be better to use a string builder here. e.g.

```
var builder = new StringBuilder ();
builder.append (element.name);
if (element.index != 0) {
    element.append ("[%u]".printf (element.index);
}
if (element.can_mute && element.is_muted) {
    builder.append (" (muted)");
} else {
    builder.append(" (%ld%%)".printf (element.volume));
}

return builder.str;
```